nsa/nsa/model.py

```
@@ -0,0 +1,1142 @@
+        dim: int,
+        block_size: int = 4,
+        num_select: int = 4,
+    ):
+        super().__init__()
+        self.block_size = block_size
+        self.num_select = num_select
+        self.dim = dim
+
+    def forward(
+        self,
+        q: torch.Tensor,
```

Ruff (0.14.8)

Unused method argument: `q`

(ARG002)